File: 백준/Silver/11441. 합 구하기/합 구하기.go
package main

import (
	"bufio"
	"fmt"
	"os"
)

func main() {
	reader := bufio.NewReader(os.Stdin)
	writer := bufio.NewWriter(os.Stdout)
	defer writer.Flush()
	var n, tc int
	var start, end int
	fmt.Fscanf(reader, "%d\n", &n)
	var numbers = make([]int, n+1)
	var numbersSum = make([]int, n+1)
	for i := 1; i <= n; i++ {
		fmt.Fscanf(reader, "%d ", &numbers[i])
	}
	numbersSum[1] = numbers[1]
	for i := 1; i <= n; i++ {
		numbersSum[i] = numbers[i] + numbersSum[i-1]
	}
	fmt.Fscanf(reader, "%d\n", &tc)
	for i := 0; i < tc; i++ {
		fmt.Fscanf(reader, "%d %d\n", &start, &end)
		fmt.Fprintf(writer, "%d\n", numbersSum[end]-numbersSum[start-1])
	}

}

File: 백준/Silver/11441. 합 구하기/합 구하기.py
import sys
input = sys.stdin.readline
sys.setrecursionlimit(10**6)
len = int(input())
numbers = list(map(int, input().split()))
numbers.insert(0,0)
numbers_sum = [0]*(len+1)
numbers_sum[1]=numbers[1]
for i in range(1, len+1):
    numbers_sum[i]=numbers[i]+numbers_sum[i-1]
tc = int(input())
for i in range(tc):
    start, end = map(int, input().split())
    print(numbers_sum[end]-numbers_sum[start-1])